lib/scitree/validations.ex

```diff
@@ -58,12 +57,12 @@ defmodule Scitree.Validations do
       iex> Scitree.Validations.validate_dataset_size(data, config)
       {:error, :incompatible_column_sizes}
   """
+  @spec validate_dataset_size(data, Config.t()) :: :ok | {:error, :incompatible_column_sizes}
   def validate_dataset_size(data, _config) do
-    {_title, _type, first} = elem(data, 0)
+    {_title, _type, first} = Enum.at(data, 0)
```

Comment:
```suggestion
    {_title, _type, first} = hd(data)
```
Since you always expect a non-empty list